### PR TITLE
special coverage rss feed

### PR DESCRIPTION
**Pull request by MichaelButkovic**
@spra85 @benghaziboy @mparent61

### CODE REVIEW

File: bulbs/feeds/views.py [hunk 1]
```diff
@@ -40,3 +43,52 @@ def get_context_data(self, *args, **kwargs):
...
+        if sc_id and sc_slug:
+            sc_ids = SpecialCoverage.objects.get(id=sc_id, slug=sc_slug).query["included_ids"]
+        elif sc_id:
+            sc_ids = SpecialCoverage.objects.get(id=sc_id).query["included_ids"]
+        elif sc_slug:
+            sc_ids = SpecialCoverage.objects.get(slug=sc_slug).query["included_ids"]
+        else:
+            sc_ids = []
+
+        return super(SpecialCoverageRSSView, self).get_queryset().filter(Ids(values=sc_ids))[:self.paginate_by]
```
Comment (benghaziboy) [inline bulbs/feeds/views.py]: custd

Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: :+1:

**Comment by benghaziboy** [comment]
will this need the same filtering of external links as I have been seeing around re: RSS feeds

**Comment by MichaelButkovic** [comment]
not as far as I know, but i'll defer to @spra85 on that

**Comment by spra85** [comment]
I wouldn't think that external link items can be set up in a special coverage list, but maybe worth confirming at least.

File: bulbs/feeds/views.py [hunk 2]
```diff
@@ -40,3 +43,52 @@ def get_context_data(self, *args, **kwargs):
...
+        return ["feeds/rss.xml", "feeds/_rss.xml"]
+
+    @cache_control(max_age=600)
+    def get(self, request, *args, **kwargs):
+        response = super(SpecialCoverageRSSView, self).get(request, *args, **kwargs)
+        response["Content-Type"] = "application/rss+xml"
+        return response
+
+    def get_queryset(self):
+        sc_id = self.request.GET.get("id")
```
Comment (spra85) [inline bulbs/feeds/views.py]: Can we prefix these parameters with something a little more descriptive?  Maybe `special_coverage_id` / `special_coverage_slug`

File: bulbs/feeds/views.py [hunk 3]
```diff
@@ -40,3 +43,52 @@ def get_context_data(self, *args, **kwargs):
...
+        sc_slug = self.request.GET.get("slug")
+
+        if sc_id and sc_slug:
+            sc_ids = SpecialCoverage.objects.get(id=sc_id, slug=sc_slug).query["included_ids"]
```
Comment (spra85) [inline bulbs/feeds/views.py]: I don't think we'd ever query on both..maybe just need either id or slug

File: bulbs/feeds/views.py (hunk 2 above)
Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: :+1: updated

File: bulbs/feeds/views.py (hunk 3 above)
Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: sounds good to me. i was just going off the story description.

File: bulbs/feeds/views.py [hunk 4]
```diff
@@ -40,3 +43,52 @@ def get_context_data(self, *args, **kwargs):
             content.feed_url = self.request.build_absolute_uri(feed_path)
 
         return RequestContext(self.request, context)
+
+
+class SpecialCoverageRSSView(ContentListView):
```
Comment (mparent61) [inline bulbs/feeds/views.py]: If you inherit from RSSView you'd avoid a lot of duplicated code.

Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: ah, yeah, good point. i'll update

**Comment by spra85** [comment]
:+1:

File: bulbs/feeds/views.py (hunk 4 above)
Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: Updated!

File: bulbs/feeds/views.py [hunk 5]
```diff
@@ -40,3 +43,21 @@ def get_context_data(self, *args, **kwargs):
...
+        sc_slug = self.request.GET.get("special_coverage_slug")
+
+        if sc_id:
+            sc_ids = SpecialCoverage.objects.get(id=sc_id).query["included_ids"]
+        elif sc_slug:
+            sc_ids = SpecialCoverage.objects.get(slug=sc_slug).query["included_ids"]
+        else:
+            sc_ids = []
+
+        return ContentListView.get_queryset(self).filter(Ids(values=sc_ids))[:self.paginate_by]
```
Comment (mparent61) [inline bulbs/feeds/views.py]: wrong base class :)

Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: oops

File: bulbs/feeds/views.py [hunk 6]
```diff
@@ -40,3 +43,21 @@ def get_context_data(self, *args, **kwargs):
...
+    def get_queryset(self):
+        sc_id = self.request.GET.get("special_coverage_id")
+        sc_slug = self.request.GET.get("special_coverage_slug")
+
+        if sc_id:
+            sc_ids = SpecialCoverage.objects.get(id=sc_id).query["included_ids"]
+        elif sc_slug:
+            sc_ids = SpecialCoverage.objects.get(slug=sc_slug).query["included_ids"]
+        else:
+            sc_ids = []
```
Comment (mparent61) [inline bulbs/feeds/views.py]: Not 404ing on missing/invalid slug/id is intentional, right? To ensure we always provide Taboola with SOME content?

Comment (MichaelButkovic) [inline bulbs/feeds/views.py]: that was the idea, yeah. if you think a 404 is better though, i can change it.

Comment (mparent61) [inline bulbs/feeds/views.py]: @spra85 ?

Comment (spra85) [inline bulbs/feeds/views.py]: I think we actually want to deliver an empty RSS in this scenario.  If they provide a slug or ID that doesn't map to anything, it should just be empty

**Comment by mparent61** [comment]
:+1:

**Pull request merged**